packages/google_identity_services_web/lib/src/js_interop/google_accounts_id.dart

```diff
@@ -85,16 +84,40 @@ extension GoogleAccountsIdExtension on GoogleAccountsId {
   ///
   /// Method: google.accounts.id.prompt
   /// https://developers.google.com/identity/gsi/web/reference/js-reference#google.accounts.id.prompt
-  external void prompt([PromptMomentListenerFn momentListener]);
+  void prompt([PromptMomentListenerFn? momentListener]) {
+    if (momentListener == null) {
+      return _prompt();
+    }
+    return _promptWithListener(momentListener.toJS);
+  }
+
+  @JS('prompt')
+  external void _prompt();
+  @JS('prompt')
+  external void _promptWithListener(JSFunction momentListener);
 
   /// Renders a Sign In With Google button in your web page.
   ///
   /// Method: google.accounts.id.renderButton
   /// https://developers.google.com/identity/gsi/web/reference/js-reference#google.accounts.id.renderButton
-  external void renderButton(
+  void renderButton(
     Object parent, [
-    GsiButtonConfiguration options,
-  ]);
+    GsiButtonConfiguration? options,
+  ]) {
+    assert(parent is JSObject,
```

Comment:
Might be cleaner to leverage type promotion here e.g.

```dart
if (parent is JSObject) {
  if (options == null) ...
  return ...
} else {
  assert(...);
}
```

Reply:
I might have to revisit this if people start encountering issues in Prod where `assert`s are removed. I don't want to have an `if` forever there.

However, since `JSObject` comes from the dart SDK, I might be able to change the type of the parent to force people to pass a JSObject (regardless of how it was obtained).